libmamba/include/mamba/core/subdirdata.hpp

```diff
@@ -22,15 +20,46 @@
 #include "mamba/core/pool.hpp"
 #include "mamba/core/repo.hpp"
 #include "mamba/core/util.hpp"
-
+#include "package_handling.hpp"
 
 namespace decompress
 {
-    bool raw(const std::string& in, const std::string& out);
+    bool raw(mamba::compression_algorithm ca, const std::string& in, const std::string& out);
 }
 
 namespace mamba
 {
+    struct subdir_metadata
+    {
+        struct checked_at
+        {
+            bool value;
+            std::time_t time;
+
+            bool has_expired() const
+            {
+                // difference in seconds, check every 14 days
+                return std::difftime(std::time(nullptr), time) > 60 * 60 * 24 * 14;
+            }
+        };
+
+        static tl::expected<subdir_metadata, std::exception> from_stream(std::istream& in);
```

Comment:
While `std::exception` is copyable, if any derived type object is stored in there, it will be sliced.
BTW would it make sense to use `mamba_error` here instead? Not sure.